src/libraries/Microsoft.PowerFx.Interpreter/Functions/SetFunction.cs

```diff
@@ -70,52 +73,58 @@ public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[
             Contracts.AssertValue(errors);
             Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
 
-            var arg0 = argTypes[0];
-
-            var firstName = args[0].AsFirstName();
+            var arg0 = argTypes[0];
+            var arg1 = argTypes[1];
 
-            if (firstName != null)
+            // Type check
+            if (!(arg0.Accepts(arg1, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: binding.Features.PowerFxV1CompatibilityRules) ||
+                 (arg0.IsNumeric && arg1.IsNumeric)))
             {
-                var info = binding.GetInfo(firstName);
-                if (info.Data is NameSymbol nameSymbol && nameSymbol.Props.CanSet)
-                {
-                    // We have a variable. type check
-                    var arg1 = argTypes[1];
+                errors.EnsureError(DocumentErrorSeverity.Critical, args[1], ErrBadType_ExpectedType_ProvidedType, arg0.GetKindString(), arg1.GetKindString());
+                return;
+            }
 
-                    if (!(arg0.Accepts(arg1, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: binding.Features.PowerFxV1CompatibilityRules) ||
-                         (arg0.IsNumeric && arg1.IsNumeric)))
+            if (arg1.AggregateHasExpandedType())
+            {
+                if (binding.Features.SkipExpandableSetSemantics)

```

Comment:
I know this code was already there, but can you add an example of a scenario where we would get this error?

Reply:
I'm not sure what this does.  :)  If you backtrack to the Feature definition, there is a comment that this was put in for Cards and is temporary and to be removed.  Anderson is following up and I opened https://github.com/microsoft/Power-Fx/issues/2461.